Source/Events.Handling/Builder/EventHandlerMethodsBuilder.cs

```diff
@@ -106,12 +146,13 @@ public EventHandlerMethodsBuilder Handle(EventTypeId eventTypeId, Generation eve
                 var eventType = eventTypes.GetFor(type);
                 if (!eventTypesToMethods.TryAdd(eventType, method))
                 {
-                    _logger.LogWarning("Event handler {EventHandler} already handles event with event type {EventType}", _eventHandlerId, eventType);
-                    HasInvalidMethods = true;
+                    _warnings.Add($"Event handler {_eventHandlerId} already handles event with event type {eventType}");
                 }
             }
 
-            return eventTypesToMethods;
+            var result = new BuildEventHandlerResult(_eventHandlerId, _warnings);
+            if (!result.Succeeded) eventTypesToMethods = null;
+            return result;
```

Comment:
I think we should just do all the checks here at the end, in stead of adding warnings as we go. Since we can't really be sure until we can resolve the eventtypes as you do here. This would simplify this code somewhat.

Reply:
This should be resolved by latest commits